Preprocessing/DataExtractionFromWiki/ModifyJson.py

```diff
@@ -1,49 +1,131 @@
+                    return {
+                        "season": int(season_str),
+                        "episode_number": ep_data.get("episode_number"),
+                        "premiere_date": ep_data.get("premiere_date"),
+                        "title": ep_data.get("title"),
+                        "viewership": ep_data.get("viewership"),
+                    }
+        return None
 
-                transcriptions_path = os.path.join(base_path, f"Sezon {season}", episode_file_name)
-                if os.path.exists(transcriptions_path):
-                    with open(transcriptions_path, "r", encoding="utf-8") as file:
-                        transcriptions = json.load(file)
+    def process_file(self, file_path: Path, episode_info: Dict[str, Any]) -> None:
+        relative_path = file_path.relative_to(self.base_path)
+        output_file_path = self.output_path / relative_path
 
-                    episode_info_data_with_season = {
-                        "season": season,
-                        **episode_info_data,
-                    }
+        episode_number = self.parse_episode_info(file_path.name)
+        if not episode_number:
+            self.logger.error(f"Filename does not match episode pattern: {file_path.name}")
+            return
 
-                    updated_transcriptions = {
-                        "episode_info": episode_info_data_with_season,
-                        "segments": transcriptions.get("segments", []),
-                    }
+        episode_info_data = self.find_episode_info(episode_info, episode_number)
+        if not episode_info_data:
+            self.logger.error(f"No episode info for Episode {episode_number}. Skipping...")
+            return
+
+        try:
+            with file_path.open("r", encoding="utf-8") as file:
+                transcriptions = json.load(file)
+
+            output_file_path.parent.mkdir(parents=True, exist_ok=True)
+            updated_transcriptions = {
+                "episode_info": episode_info_data,
+                "segments": transcriptions.get("segments", []),
+            }
+
+            with output_file_path.open("w", encoding="utf-8") as file:
+                json.dump(updated_transcriptions, file, ensure_ascii=False, indent=4)
+
+            self.logger.info(f"Created file: {output_file_path}")
+
+        except FileNotFoundError as exc:
+            self.logger.error(f"File not found: {file_path}. Error: {exc}")
+            raise ValueError(f"File not found: {file_path}") from exc
+        except json.JSONDecodeError as exc:
+            self.logger.error(f"Error decoding JSON in file {file_path}: {exc}")
+            raise ValueError(f"Error decoding JSON in file {file_path}: {exc}") from exc
+        except PermissionError as exc:
+            self.logger.error(f"Permission denied when accessing file: {file_path}. Error: {exc}")
+            raise ValueError(f"Permission denied when accessing file: {file_path}") from exc
```

Comment:
```suggestion
        except FileNotFoundError as exc:
            raise ValueError(f"File not found: {file_path}") from exc
        except json.JSONDecodeError as exc:
            raise ValueError(f"Error decoding JSON in file {file_path}: {exc}") from exc
        except PermissionError as exc:
            raise ValueError(f"Permission denied when accessing file: {file_path}") from exc
```

ten sam case z tym exceptionem